src/otx/cli/cli.py

```diff
@@ -0,0 +1,316 @@
+                nested_key="scheduler",
+                **scheduler_kwargs,
+            )
+
+            skip: set[str | int] = set(self.engine_subcommands()[subcommand])
+            fn = getattr(Engine, subcommand)
+            description = get_short_docstring(fn)
+            added = sub_parser.add_method_arguments(
+                Engine,
+                subcommand,
+                skip=skip,
+                fail_untyped=False,
+            )
+
+            # Load default subcommand config file
+            default_config_file = Path(get_otx_root_path()) / "recipe" / "_base_" / f"{subcommand}.yaml"
+            if default_config_file.exists():
+                with Path(default_config_file).open() as f:
+                    default_config = yaml.safe_load(f)
+                sub_parser.set_defaults(**default_config)
+
+            if "logger" in added:
+                sub_parser.link_arguments("engine.work_dir", "logger.init_args.save_dir")
+            if "callbacks" in added:
+                sub_parser.link_arguments("callback_monitor", "callbacks.init_args.monitor")
+                sub_parser.link_arguments("engine.work_dir", "callbacks.init_args.dirpath")
+
+            self._subcommand_method_arguments[subcommand] = added
+            self._subcommand_parsers[subcommand] = sub_parser
+            parser_subcommands.add_subcommand(subcommand, sub_parser, help=description)
+
+    def _set_extension_subcommands_parser(self, parser_subcommands: _ActionSubCommands) -> None:
+        from otx.cli.install import add_install_parser
+
+        add_install_parser(parser_subcommands)
+
+    def instantiate_classes(self) -> None:
+        """Instantiate the necessary classes based on the subcommand.
+
+        This method checks if the subcommand is one of the engine subcommands.
+        If it is, it instantiates the necessary classes such as config, datamodule, model, and engine.
+        """
+        if self.subcommand in self.engine_subcommands():
+            self.config_init = self.parser.instantiate_classes(self.config)
+            self.datamodule = self._get(self.config_init, "data")
+            self.model, optimizer, scheduler = self.instantiate_model()
+
+            engine_kwargs = self._get(self.config_init, "engine")
+            self.engine = Engine(
+                model=self.model,
+                optimizer=optimizer,
+                scheduler=scheduler,
+                datamodule=self.datamodule,
+                **engine_kwargs,
+            )
+
+    def instantiate_model(self) -> tuple:
+        """Instantiate the model based on the subcommand.
+
+        This method checks if the subcommand is one of the engine subcommands.
+        If it is, it instantiates the model.
+
+        Returns:
+            tuple: The model and optimizer and scheduler.
+        """
+        model = self._get(self.config_init, "model")
+        optimizer_kwargs = self._get(self.config_init, "optimizer", {})
```

Comment:
What's the type of `self.config_init` and why `optimizer_kwargs` has more than two types (`Namespace, dict, ...?`)?

Reply:
When `parse_args` in `argparser`, config is instantiated as a nested dictionary, so I modified it to accept a `dict` type.
Actually, it would be fine to modify it just for that class (the class that uses a dict argument named config in the config file// + OVModel), but for now I did the same for the model class, which uses config in the same way.

Reply:
This looks like a mistake on my part. I've fixed it to communicate with Namespace. Thanks! - https://github.com/openvinotoolkit/training_extensions/pull/2805/commits/1a254d9d1306b9389b89a5594eefa8faf43f2e38